README.md

```diff
@@ -22,6 +22,48 @@ The naive, [highway hash](https://github.com/google/highwayhash) based PSI: an *
 
 The [bloomfilter](https://en.wikipedia.org/wiki/Bloom_filter) based PSI: an *insecure* but fast with lower communication overhead than [npsi](pkg/npsi/README.md) solution for PSI. Take a look [here](pkg/bpsi/README.md) to consult the documentation.
 
+## logging
+
+`github.com/go-logr/logr` is used internally for logging, which accepts a `logr.Logger` object. See the documentation on `logr` for various implementation of `logr.Logger`. Example implementation of match sender and receiver uses `github.com/go-logr/stdr` which logs to `os.Stderr`.
+
+### pass logger to sender or receiver
+To pass a logger to a sender or a receiver, create a new context with the parent context and `logr.Logger` object as follows
+```golang
+// for new sender with logger
+ctx := logr.NewContext(parentCtx, logger)
+sender, err := psi.NewSender(psiType, ctx, conn)
```

Comment:
The doc needs to be adjusted. We pass the context to the Send function, not to NewSender. Idem for receiver.